### PR TITLE
Fix Prophet wrapper to work with `Timestamp` instead of `Period`

### LINKED ISSUE #2181: Prophet not compatible with `Period`
## Description
TypeError is raised when using a PandasDataset with the ProphetPredictor.

The problem seems to come from the `pd.to_datetime` function which does not accept Period. I guess it broke when this was done: https://github.com/awslabs/gluon-ts/pull/1980

## To Reproduce
Using a simple PandasDataset than can be found [here](https://ts.gluon.ai/v0.10.x/tutorials/data_manipulation/pandasdataframes.html#Use-case-1---Loading-data-from-a-long-dataframe) we can reproduce the error easily:

```python
import pandas as pd
from gluonts.dataset.pandas import PandasDataset
from gluonts.model.prophet import ProphetPredictor

# Load Data
url = (
    "https://gist.githubusercontent.com/rsnirwan/a8b424085c9f44ef2598da74ce43e7a3"
    "/raw/b6fdef21fe1f654787fa0493846c546b7f9c4df2/ts_long.csv"
)
df = pd.read_csv(url, index_col=0, parse_dates=True)

# Create Dataset
ds = PandasDataset.from_long_dataframe(df, target="target", item_id="item_id")

# Init predictor
predictor = ProphetPredictor(prediction_length=1)

# Run forecast
next(predictor.predict(ds))
```

## Error message or code output
(Paste the complete error message, including stack trace, or the undesired output that the above snippet produces.)

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
----> 1 next(predictor.predict(ds))

~/.venv/lib/python3.7/site-packages/gluonts/model/prophet/_predictor.py in predict(self, dataset, num_samples, **kwargs)
    155             data = self._make_prophet_data_entry(entry)
    156 
--> 157             forecast_samples = self._run_prophet(data, params)
    158 
    159             yield SampleForecast(

~/.venv/lib/python3.7/site-packages/gluonts/model/prophet/_predictor.py in _run_prophet(self, data, params)
    174             prophet.add_regressor(feat_name(i))
    175 
--> 176         prophet.fit(data.prophet_training_data)
    177 
    178         future_df = prophet.make_future_dataframe(

~/.venv/lib/python3.7/site-packages/prophet/forecaster.py in fit(self, df, **kwargs)
   1112         if history.shape[0] < 2:
   1113             raise ValueError('Dataframe has less than 2 non-NaN rows.')
-> 1114         self.history_dates = pd.to_datetime(pd.Series(df['ds'].unique(), name='ds')).sort_values()
   1115 
   1116         history = self.setup_dataframe(history, initialize_scales=True)

~/.venv/lib/python3.7/site-packages/pandas/core/tools/datetimes.py in to_datetime(arg, errors, dayfirst, yearfirst, utc, format, exact, unit, infer_datetime_format, origin, cache)
    885             result = arg.map(cache_array)
    886         else:
--> 887             values = convert_listlike(arg._values, format)
    888             result = arg._constructor(values, index=arg.index, name=arg.name)
    889     elif isinstance(arg, (ABCDataFrame, abc.MutableMapping)):

~/.venv/lib/python3.7/site-packages/pandas/core/tools/datetimes.py in _convert_listlike_datetimes(arg, format, name, tz, unit, errors, infer_datetime_format, dayfirst, yearfirst, exact)
    364     orig_arg = arg
    365     try:
--> 366         arg, _ = maybe_convert_dtype(arg, copy=False)
    367     except TypeError:
    368         if errors == "coerce":

~/.venv/lib/python3.7/site-packages/pandas/core/arrays/datetimes.py in maybe_convert_dtype(data, copy)
   2260         #  test_setops.test_join_does_not_recur fails
   2261         raise TypeError(
-> 2262             "Passing PeriodDtype data is invalid. Use `data.to_timestamp()` instead"
   2263         )
   2264 

TypeError: Passing PeriodDtype data is invalid. Use `data.to_timestamp()` instead
```

## Environment
- Operating system: WSL2 - Ubuntu
- Python version: 3.7.12
- GluonTS version: 0.10.2
- MXNet version: 1.8.0.post0 (cu110)
- Prophet version: 1.1
- Pandas version: 1.3.5

### PR DESCRIPTION
**Pull request by lostella**
*Issue #, if available:* fixes #2181

*Description of changes:* The fix does what is asked by Prophet (see stack trace in #2181). However it is strange that tests using `ListDataset` do not raise this issue.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

**Review by jaheba — DISMISSED** [review]

**Comment by jaheba** [comment]
Weird indeed that it wasn't caught.

Should we add another test?

**Comment by lostella** [comment]
@jaheba seems like different frequencies in the existing tests would surface the issue also with `ListDataset`; added test cases (now passing).

**Comment by jaheba** [comment]
We need to enable prophet tests by installing `prophet` in one of the tests workflow.

File: test/model/prophet/test_prophet.py [hunk 1]
```diff
@@ -41,17 +50,17 @@ def test_feat_dynamic_real_success():
...
 
     act_fcst = next(predictor.predict(dataset))
     exp_fcst = np.arange(5.0, 5.0 + params["prediction_length"])
 
-    assert np.all(np.isclose(act_fcst.quantile(0.1), exp_fcst, atol=0.02))
-    assert np.all(np.isclose(act_fcst.quantile(0.5), exp_fcst, atol=0.02))
-    assert np.all(np.isclose(act_fcst.quantile(0.9), exp_fcst, atol=0.02))
+    assert exp_fcst.shape == act_fcst.quantile(0.1).shape
+    assert exp_fcst.shape == act_fcst.quantile(0.5).shape
+    assert exp_fcst.shape == act_fcst.quantile(0.9).shape
```
Comment (lostella) [inline test/model/prophet/test_prophet.py:63]: The previous assertions were very strict, and some started failing with `prophet==1.1`; I opted for making this test not about Prophet's accuracy, but rather about the wrapper's behavior.

Concerns @jaheba?

Comment (jaheba) [reply test/model/prophet/test_prophet.py:63]: Interesting that it gets worse. I think this makes sense, as we can't guarantee prophets results but merely provide a wrapper.

**Review by jaheba — APPROVED** [review]

**Comment by kashif** [comment]
cool! nice catch!

**Pull request merged**